application/org.openjdk.jmc.jolokia/src/main/java/org/openjdk/jmc/jolokia/JolokiaDiscoveryListener.java

```diff
@@ -49,10 +47,13 @@
  */
 public class JolokiaDiscoveryListener extends AbstractCachedDescriptorProvider implements PreferenceConstants {
 
-	JolokiaDiscoverySettings settings;
+	private JolokiaDiscoverySettings settings;
+	private final JolokiaDiscovery jolokiaDiscovery;
 
 	public JolokiaDiscoveryListener(JolokiaDiscoverySettings settings) {
 		this.settings = settings;
+		jolokiaDiscovery = new JolokiaDiscovery();
```

Comment:
Ensure that only one instance is instantiated (linked to the lifecycle of this plugin). If one tries to instantiate another, it will attempt to register an MBean and sinply crash